src/alphabets/mod.rs

```diff
@@ -152,6 +152,72 @@ impl Alphabet {
     pub fn is_empty(&self) -> bool {
         self.symbols.is_empty()
     }
+
+    /// Return a new alphabet taking the intersect between this and other.
+    ///
+    /// # Example
+    /// ```
+    /// use bio::alphabets;
+    ///
+    /// let alpha_a = alphabets::Alphabet::new(b"acgtACGT");
+    /// let alpha_b = alphabets::Alphabet::new(b"atcgMVP");
+    /// let intersect_alpha = alpha_a.intersection(alpha_b);
+    ///
+    /// assert_eq!(intersect_alpha, alphabets::Alphabet::new(b"atcg"));
+    /// ```
+    pub fn intersection(self, other: Alphabet) -> Self {
+        let mut new_symbols = BitSet::new();
+        for symbol_to_add in self.symbols.intersection(&other.symbols) {
+            new_symbols.insert(symbol_to_add);
+        }
+        return Alphabet {
+            symbols: new_symbols,
+        };
+    }
+
+    /// Return a new alphabet taking the difference between this and other.
+    ///
+    /// # Example
+    /// ```
+    /// use bio::alphabets;
+    ///
+    /// let dna_alphabet = alphabets::Alphabet::new(b"acgtACGT");
+    /// let dna_alphabet_upper = alphabets::Alphabet::new(b"ACGT");
+    /// let dna_lower = dna_alphabet.difference(dna_alphabet_upper);
+    ///
+    /// assert_eq!(dna_lower, alphabets::Alphabet::new(b"atcg"));
+    /// ```
+    pub fn difference(self, other: Alphabet) -> Self {
+        let mut new_symbols = BitSet::new();
+        for symbol_to_add in self.symbols.difference(&other.symbols) {
+            new_symbols.insert(symbol_to_add);
+        }
+        return Alphabet {
+            symbols: new_symbols,
+        };
```

Comment:
Same suggestion as for intersection
```suggestion
        Alphabet {
            symbols: self.symbols.difference(&other.symbols).collect(),
        }
```